src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (!(prey instanceof Herbivore)
+      || prey.hidden
+    ) {
+      return;
+    }
+
+    prey.health -= 50;
+
+    Animal.alive = Animal.alive.filter(
+      ({ health }) => health > 0
+    );
```

Comment:
do we need to do it if the health of the animal is not lower than 0?

Reply:
the task said that animal should be removed when it's health reaches zero. So yeah